internal/tracefs/kprobe.go

```diff
@@ -113,6 +113,10 @@ func sanitizeTracefsPath(path ...string) (string, error) {
 // but may be also be available at /sys/kernel/debug/tracing if debugfs is mounted.
 // The available tracefs paths will depends on distribution choices.
 var getTracefsPath = sync.OnceValues(func() (string, error) {
+	if runtime.GOOS != "linux" {
```

Comment:
I wonder if this helper would make sense:
```go
package internal/linux

func Feature(feat string) error {
	if runtime.GOOS == "linux" {
		return nil
	}
	return fmt.Errorf("%s: %w", feat, internal.ErrNotSupportedOnOS)
}
```

used like:

```go
if err := linux.Feature("tracefs"); err != nil {
	return "", err
}
```

That would reduce the sprawl of open coding `runtime.GOOS != "linux"`, since the majority of cases (for now) will check if running on Linux specifically. Also no need for wrapping the error, and all calls will be easily discoverable using reference search. I'm open to other ideas and bike shedding, of course. :slightly_smiling_face:

Reply:
I'm not sure this is clearer than just open coding if runtime.GOOS != ... in the code base. It's not shorter and also not easier to discover, since one can always find references of runtime.GOOS or grep for the condition. Idle thought: I could add a constant like this https://go.dev/play/p/_JOiIsEduDo if you really dislike the open coding.

`if internal.IsWindows { ... }`

I like the idea of having something like this in testutils though: testutils.LinuxOnly(reason) or similar. That forces me to document why a test was stubbed out.

Reply:
P.S. the bits where `runtime.GOOS == "windows"` come later :P